src/Utility/ConstraintStabiliser.php

```diff
@@ -15,6 +15,36 @@ class ConstraintStabiliser
     private const GRAPHQL_3_VERSION_DEV = '3.8.x-dev';
     private const GRAPHQL_3_VERSION_STABLE = '~3.8.1@stable';
 
+    private static function getRootLibraryRelease(LibraryRelease $libraryRelease): LibraryRelease
+    {
+        $parentRelease = $libraryRelease->getParentRelease();
+        if ($parentRelease) {
+            return self::getRootLibraryRelease($parentRelease);
+        }
+        return $libraryRelease;
+    }
+
+    private static function getItems(LibraryRelease $libraryRelease, array &$names, array &$items)
+    {
+        foreach ($libraryRelease->getAllItems() as $item) {
+            $name = $item->getLibrary()->getName();
+            if (!in_array($name, $names)) {
+                $items[] = $item;
+                $names[] = $name;
+            }
+            self::getItems($item, $names, $items);
+        }
+        return $items;
+    }
+
+    private static function getAllItems(LibraryRelease $libraryRelease)
+    {
+        $names = [];
+        $items = [];
+        $rootLibraryRelease = self::getRootLibraryRelease($libraryRelease);
+        return self::getItems($rootLibraryRelease, $names, $items);
+    }
```

Comment:
Remove these methods - all we need is to call `$libraryRelease->getRootRelease()->getAllItems();`

Reply:
Updated